### ISSUE TITLE
Allow the usage of both Arabic and Hindu-Arabic numbers in any "ar-*" locale

### ISSUE THREAD

**Issue by mahmoudajawad** [robisim74/angular-l10n#161]
**I'm submitting a...**
```
[x] bug report
[ ] feature request
[ ] support request
```

**Expected behavior**
As Arabs are using two different numerical systems, the Arabic (1, 2, 3..) and the Hindu-Arabic (١, ٢, ٣, ٤..) I would expect the validation of the numbers to be allowed in both the set of numberic systems.

**Actual behavior**
Cuurently, angular-l10n only validates the Hindu-Arabic numbers.

**Steps to reproduce the behavior**
- Go to: https://robisim74.github.io/angular-l10n-sample/#/validation
- Change locale to Saudi Arabia (the only Arabic locale in the demo)
- Try inputting (1,000.00) for validation

**Specifications: Angular 5, library 4.1.4, environment Windows 10 (WSL), browser Firefox, Chrome**

**Comment by robisim74** [comment]
Hi @AjawadMahmoud,

this library manages one `defaultLocale` at a time, even for validation: it is a design choice, not a bug.

You can change the numbering system when you call the `setDefaultLocale` method:

```
this.locale.setDefaultLocale('ar', 'SA, '', 'arab'); // 'arab' is the default => Hindu-Arabic
this.locale.setDefaultLocale('ar', 'SA, '', 'latn'); // 'latn' => Arabic
```

This said, if you want to enable the user to insert two different types of numbering _at the same time_, you could build a custom validation directive, something like this:

```
export function MyValidateNumber(): ValidatorFn {

    return (formControl: FormControl): ValidationErrors | null => {
        const l10nControl: FormControl = new FormControl(formControl.value, [l10nValidateNumber('1.2-2')]);

        const valid: boolean = l10nControl.valid || /^\d+\.\d{2,2}$/.test(formControl.value); // your alternative validation logic

        return !valid ? { myValidation: { value: false } } : null;
    };

}
```

Then, if you have any ideas on how to improve the library directive (passing the `defaultLocale`? passing an alternative numbering system?), let me know.

**Comment by mahmoudajawad** [comment]
Hello,,

I'm sorry for calling it a bug at first.
Your solution is so great, it's even future-proof if I start using your lib. But, since you are open (thanks!) to ideas, waht about something like:
`
this.locale.setDefaultLocale('ar', 'SA, '', 'latn-arab'); // a combination of both numberic systems for validation with Arabic (latn) numbers for display
this.locale.setDefaultLocale('ar', 'SA, '', 'arab-latn'); // a combination of both numberic systems for validation with Hindu-Arabic (arab) numbers for display
`
basically, adding additional options (over changing current behaviour) would allow everybody to use their preferred validation and display options at all time, without breaking current apps using the lib.
how does it look?
if i have time today i'll try to implement it locally. if it works i'd fork your repo and create a PR. just pray i get time, please xD.

**Comment by robisim74** [comment]
Passing an alternative numbering system during the configuration seems a good idea, but you can't do a _combination_, because it will break the IntlAPI: maybe it should be an alternative parameter for the numbering system for each locale configuration to use as alternative in particular situations, like the validation.

Then, doing the validation that accepts two numbering systems will require to have _two instances_ of the validation algorithm (it creates a specific regex depending on the _defaultLocale_).

I was already thinking of doing a refactoring of the `LocaleValidation `class (merging the others classes for validation into it), and to make sure that the validation regex depends only on the _defaultLocale_ and that is not calculated by each `l10nValidateNumber `directive (performance improvement when inserting many numbers on the same page).

**Comment by robisim74** [comment]
@AjawadMahmoud I refactored the `LocaleValidation` class and I added a different numbering system in the sample app as I said in the comment above: https://github.com/robisim74/angular-l10n/issues/161#issuecomment-356036272

I do not see it possible for the library to work with multiple numbering systems at the same time.

If you have any ideas, feel free to reopen this issue or make proposals.

Greetings
